torch/_dynamo/graph_region_tracker.py

```diff
@@ -0,0 +1,225 @@
+import functools
+import math
+from collections import defaultdict, deque
+from typing import Any, Callable, Deque, Dict, List, Optional, Set, Tuple
+
+import torch.fx
+from torch.utils._pytree import tree_flatten
+
+
+Node = torch.fx.Node
+Region = List[Node]
+IdenticalNodes = List[Node]
+GlobalStateKey = Tuple[bool, bool, int, bool, bool, torch.dtype, bool, bool, bool]
+NodeKey = Tuple[GlobalStateKey, str, int]
+
+
+def _get_tensor_props(node: Node) -> Optional[Tuple[torch.Size, torch.dtype]]:
+    if isinstance(node, torch.fx.Node):
+        value = node.meta.get("example_value", None)
+        if isinstance(value, torch.Tensor):
+            return value.shape, value.dtype
+    return None
```

Comment:
I think this is missing a lot. We may need strides, device, layout, requires_grad-ness and Tensor subclassness among other properties

Reply:
I updated this to use the `extract_tensor_metadata_for_cache_key` from the inductor codecache.